magenta/models/melody_rnn/melody_rnn_sequence_generator.py

```diff
@@ -155,19 +155,23 @@ def _generate(self, input_sequence, generator_options):
     # Ensure that the melody extends up to the step we want to start generating.
     melody.set_length(start_step - melody.start_step)
 
+    temperature = (generator_options.args['temperature'].int_value
+                   if 'temperature' in generator_options.args else 1.0)
     generated_melody = self.generate_melody(
-        end_step - melody.start_step, melody)
+        end_step - melody.start_step, melody, temperature)
     generated_sequence = generated_melody.to_sequence(qpm=qpm)
-    assert generated_sequence.total_time <= generate_section.end_time
+    assert (generated_sequence.total_time - generate_section.end_time) <= 1e-5
```

Comment:
Unrelated but needs to be fixed since I kept hitting the assertion when testing.